packages/jellyfish-api-core/__tests__/category/icxorderbook/submitDFCHTLC.test.ts

```diff
@@ -0,0 +1,338 @@
+    // maker deposit should be reduced from accountDFI
+    expect(accountDFIAfterExtHTLC[idDFI]).toStrictEqual(accountDFIBeforeExtHTLC[idDFI].minus(0.1))
+
+    const accountBTCBeforeDFCHTLC: Record<string, BigNumber> = await client.call('getaccount', [accountBTC, {}, true], 'bignumber')
+    // create DFCHTLC - taker
+    const DFCHTLC: HTLC = {
+      offerTx: makeOfferTxId,
+      amount: new BigNumber(100), // in  DFC
+      hash: '957fc0fd643f605b2938e0631a61529fd70bd35b2162a21d978c41e5241a5220'
+    }
+    const DFCHTLCTxId = (await client.icxorderbook.submitDFCHTLC(DFCHTLC)).txid
+    await container.generate(1)
+
+    const accountBTCAfterDFCHTLC: Record<string, BigNumber> = await client.call('getaccount', [accountBTC, {}, true], 'bignumber')
+    expect(accountBTCAfterDFCHTLC[idDFI]).toStrictEqual(accountBTCBeforeDFCHTLC[idDFI].minus(100))
+
+    // List htlc
+    const listHTLCOptionsAfterDFCHTLC = {
+      offerTx: makeOfferTxId
+    }
+    const HTLCsAfterDFCHTLC: Record<string, ICXDFCHTLCInfo | ICXEXTHTLCInfo | ICXClaimDFCHTLCInfo> = await client.call('icx_listhtlcs', [listHTLCOptionsAfterDFCHTLC], 'bignumber')
+    expect(Object.keys(HTLCsAfterDFCHTLC).length).toBe(3) // extra entry for the warning text returned by the RPC atm.
+    expect(HTLCsAfterDFCHTLC[DFCHTLCTxId] as ICXDFCHTLCInfo).toStrictEqual(
+      {
+        type: ICXHTLCType.DFC,
+        status: ICXOrderStatus.OPEN,
+        offerTx: makeOfferTxId,
+        amount: DFCHTLC.amount,
+        amountInEXTAsset: DFCHTLC.amount.dividedBy(order.orderPrice),
+        hash: DFCHTLC.hash,
+        timeout: new BigNumber(500),
+        height: expect.any(BigNumber),
+        refundHeight: expect.any(BigNumber)
+      }
+    )
+  })
+
+  it('should submit DFC HTLC with input utxos for a DFC buy offer', async () => {
+    const { order, createOrderTxId } = await icxSetup.createDFISellOrder('BTC', accountDFI, '037f9563f30c609b19fd435a19b8bde7d6db703012ba1aba72e9f42a87366d1941', new BigNumber(15), new BigNumber(0.01))
+    const { makeOfferTxId } = await icxSetup.createDFIBuyOffer(createOrderTxId, new BigNumber(0.10), accountBTC)
+
+    const accountDFIBeforeDFCHTLC: Record<string, BigNumber> = await client.call('getaccount', [accountDFI, {}, true], 'bignumber')
+    // create DFCHTLC - maker
+    const DFCHTLC: HTLC = {
+      offerTx: makeOfferTxId,
+      amount: new BigNumber(10), // in  DFC
+      hash: '957fc0fd643f605b2938e0631a61529fd70bd35b2162a21d978c41e5241a5220',
+      timeout: 500
+    }
+
+    // input utxos
+    const utxos = await container.call('listunspent', [1, 9999999, [accountDFI], true])
+    const inputUTXOs: UTXO[] = utxos.map((utxo: UTXO) => {
+      return {
+        txid: utxo.txid,
+        vout: utxo.vout
+      }
+    })
+    const DFCHTLCTxId = (await client.icxorderbook.submitDFCHTLC(DFCHTLC, inputUTXOs)).txid
+    await container.generate(1)
```

Comment:
```suggestion
    const DFCHTLCTxId = (await client.icxorderbook.submitDFCHTLC(DFCHTLC, inputUTXOs)).txid
    await container.generate(1)
    
    const rawtx = await container.call('getrawtransaction', [DFCHTLCTxId, true])
    // here assert the rawtx.vin whether contains the inputUTXOs
```
gonna assert whether the inputUTXOs are really be inserted into tx.vin
(this my my bad utxos input test pattern previously, will update those)

Reply:
the easy way is to use `fundAddress` where you will be given a set of UTXO that is ready to be used as listutxos might not give you an UTXO for use

https://github.com/DeFiCh/jellyfish/blob/78f2509421c86e29da77c771cd6371044f5e5fbb/packages/jellyfish-api-core/__tests__/category/oracle/setOracleData.test.ts#L113